editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/maven/CamelProjectConfigurator.java

```diff
@@ -174,6 +173,28 @@ private boolean isWARProject(IProject project, IProgressMonitor monitor) {
 		return CamelProjectConfigurator.WAR_PACKAGE.equalsIgnoreCase(m2prj.getPackaging());
 	}
 
+	private boolean isInvalidCamelFacetVersion(ProjectConfigurationRequest request) throws CoreException {
+		IProject project = request.getProject();
+		IFacetedProject fproj = ProjectFacetsManager.create(project);
+		if (fproj != null) {
+			return !fproj.getInstalledVersion(camelFacet).getVersionString().equalsIgnoreCase(DEFAULT_CAMEL_FACET_VERSION);
+		}
+		return true;
+	}
+	
+	private void repairCamelFacet(ProjectConfigurationRequest request) throws CoreException {
+		IProject project = request.getProject();
+		IFacetedProject fproj = ProjectFacetsManager.create(project);
+		NullProgressMonitor mon = new NullProgressMonitor();
```

Comment:
better to provide the monitor as a parameter to provide a better feedback to the user